src/module/rules/rule-element/grant-item/rule-element.ts

```diff
@@ -321,12 +347,11 @@ class GrantItemRuleElement extends RuleElementPF2e<GrantItemSchema> {
                 (setHasElement(PHYSICAL_ITEM_TYPES, grantee.type) ? "detach" : "cascade"),
         };
 
-        if (grantee instanceof ItemPF2e) {
+        grantee.flags = fu.mergeObject(grantee.flags ?? {}, { pf2e: { grantedBy } });
+        if (grantee instanceof ItemPF2e && grantee._id) {
```

Comment:
Check that the actor has the ID in its embedded collection rather than just confirming it has an ID.

Reply:
I can do that, though the reason I did that check was actually so that typescript wouldn't complain. I figured if the original didn't check, the new one didn't, but there's harm in actually checking.